src/__tests__/ParserComponent.test.tsx

```diff
@@ -61,22 +62,214 @@ describe('Csv FileEvents processed correctly', () => {
+      );
+      const fileEvent: any = {target: {files: [pdfTestFile]}};
+      // try {
+      //   // todo: make sure it throws the right error
+      //   expect(await comp.instance().parse(fileEvent)).toThrowError();
+      // } catch (error) {
+      //   console.log(error);
+      // }
+    });
+
+    // Currently failing because no error handling
+    it('.txt rejected', async () => {
+      const txtTestFile: File = new File(
+          ['abcdef'],
+          'test.txt',
+          {type: '.txt, text/plain'},
+      );
+      const fileEvent: any = {target: {files: [txtTestFile]}};
+      // try {
+      //   // todo: make sure it throws the right error
+      //   expect(await comp.instance().parse(fileEvent)).toThrowError();
+      // } catch (error) {
+      //   console.log(error);
+      // }
+    });
+
+    it('.doc rejected', async () => {
+      const docTestFile: File = new File(
+          [''],
+          'test.doc',
+          {type: '.doc, application/msword'},
+      );
+      const fileEvent: any = {target: {files: [docTestFile]}};
+      // try {
+      //   // todo: make sure it throws the right error
+      //   expect(await comp.instance().parse(fileEvent)).toThrowError();
+      // } catch (error) {
+      //   console.log(error);
+      // }
+    });
+
+    it('.css rejected', async () => {
+      const cssTestFile: File = new File(
+          [''],
+          'test.css',
+          {type: '.css, text/css'},
+      );
+      const fileEvent: any = {target: {files: [cssTestFile]}};
+      // try {
+      //   // todo: make sure it throws the right error
+      //   expect(await comp.instance().parse(fileEvent)).toThrowError();
+      // } catch (error) {
+      //   console.log(error);
+      // }
+    });
+
+    it('.js rejected', async () => {
+      const jsTestFile: File = new File(
+          ['abcdefg'],
+          'test.js',
+          {type: '.js, text/javascript'},
+      );
+      const fileEvent: any = {target: {files: [jsTestFile]}};
+      // try {
+      //   // todo: make sure it throws the right error
+      //   expect(await comp.instance().parse(fileEvent)).toThrowError();
+      // } catch (error) {
+      //   console.log(error);
+      // }
     });
   });
 
-  describe('componentDidMount()', () => {
-    it('dummy test', () => {
-      // todo: devs need to write unit tests
+  it('.csv file with no temporal data is rejected', async () => {
+    const onChangeMock = jest.fn();
+    const noDateFile: File = new File(
+        ['H1,H2,H3,H4\n' +
+                'a,b,c,d\n' +
+                'e,f,g,h\n' +
+                'i,j,k,l\n'],
+        'test.csv',
+        {type: '.csv,text/csv'},);
+    const comp: any = mount(<ParserComponent
+      {...props}
+      onChange={onChangeMock}
+    />);
+    const fileEvent = {target: {files: [noDateFile]}};
+    try {
+      expect(
+          await comp.instance().parse(fileEvent)
+      ).toThrowError('The file uploaded has no dates.');
```

Comment:
Nice, this is how I'd expect the error to be checked. However are you sure you need to wrap it in a try catch?